File: test/samples/vulnerable-app.js
// Vulnerable JavaScript Application - Test Sample
// Contains quantum-vulnerable cryptographic patterns for E2E testing

const crypto = require('crypto');

// RSA 2048-bit key generation (quantum-vulnerable)
function generateRsaKeyPair() {
    const { publicKey, privateKey } = crypto.generateKeyPairSync('rsa', {
        modulusLength: 2048,
        publicKeyEncoding: { type: 'spki', format: 'pem' },
        privateKeyEncoding: { type: 'pkcs8', format: 'pem' }
    });
    return { publicKey, privateKey };
}

// ECDSA signing (quantum-vulnerable)
function signWithEcdsa(data, privateKey) {
    const sign = crypto.createSign('SHA256');
    sign.update(data);
    return sign.sign(privateKey, 'hex');
}

// ECDH key exchange (quantum-vulnerable)
function performEcdh() {
    const ecdh = crypto.createECDH('secp256k1');
    ecdh.generateKeys();
    return ecdh.getPublicKey('hex');
}

// MD5 hash (weak algorithm)
function hashWithMd5(data) {
    return crypto.createHash('md5').update(data).digest('hex');
}

// SHA-1 hash (deprecated)
function hashWithSha1(data) {
    return crypto.createHash('sha1').update(data).digest('hex');
}

// DES encryption (weak algorithm)
function encryptWithDes(data, key) {
    const cipher = crypto.createCipher('des', key);
    return cipher.update(data, 'utf8', 'hex') + cipher.final('hex');
}

// 3DES encryption (deprecated)
function encryptWith3des(data, key) {
    const cipher = crypto.createCipher('des-ede3', key);
    return cipher.update(data, 'utf8', 'hex') + cipher.final('hex');
}

module.exports = {
    generateRsaKeyPair,
    signWithEcdsa,
    performEcdh,
    hashWithMd5,
    hashWithSha1,
    encryptWithDes,
    encryptWith3des
};

File: test/samples/vulnerable-app.py
# Vulnerable Python Application - Test Sample
# Contains quantum-vulnerable cryptographic patterns for E2E testing

from cryptography.hazmat.primitives.asymmetric import rsa, dsa, ec
from cryptography.hazmat.primitives import hashes
from cryptography.hazmat.backends import default_backend
import hashlib

# RSA 2048-bit key generation (quantum-vulnerable)
def generate_rsa_key():
    private_key = rsa.generate_private_key(
        public_exponent=65537,
        key_size=2048,
        backend=default_backend()
    )
    return private_key

# RSA 1024-bit key (weak and quantum-vulnerable)
def generate_weak_rsa_key():
    private_key = rsa.generate_private_key(
        public_exponent=65537,
        key_size=1024,
        backend=default_backend()
    )
    return private_key

# DSA key generation (quantum-vulnerable)
def generate_dsa_key():
    private_key = dsa.generate_private_key(
        key_size=2048,
        backend=default_backend()
    )
    return private_key

# ECDSA key generation (quantum-vulnerable)
def generate_ecdsa_key():
    private_key = ec.generate_private_key(
        ec.SECP256K1(),
        backend=default_backend()
    )
    return private_key

# MD5 hash (weak algorithm)
def hash_md5(data):
    return hashlib.md5(data.encode()).hexdigest()

# SHA-1 hash (deprecated)
def hash_sha1(data):
    return hashlib.sha1(data.encode()).hexdigest()

# DES encryption import (weak algorithm)
from Crypto.Cipher import DES

def encrypt_des(data, key):
    cipher = DES.new(key, DES.MODE_ECB)
    return cipher.encrypt(data)

# 3DES encryption import (deprecated)
from Crypto.Cipher import DES3

def encrypt_3des(data, key):
    cipher = DES3.new(key, DES3.MODE_ECB)
    return cipher.encrypt(data)
